openvm/src/lib.rs

```diff
@@ -527,63 +521,6 @@ pub fn compile_openvm(
     })
 }
 
-pub fn compile_guest(
```

Comment:
Removed this function, users have to call `compile_openvm` and `compile_exe`. We had a few places where we called `compile_openvm` several times, where now we can just call it once and pass the result.